conversation.pl

```diff
@@ -0,0 +1,45 @@
+prompt_conversation :-
+	position(hero, Location),
+	npc_position(NPC, Location),
+	game_in_progress,
+	state(NPC, 1, 0).
+
+prompt_conversation :-
+	position(hero, Location),
+	\+ npc_position(_, Location),
+	write("There is no one around..."), nl.
+
+	
+state(npc_1, 1, 0) :-
```

Comment:
Maybe rename it to dialog_state or sth like this. Naming states with numbers makes the code rather obscure. Anyway, I'm not really sure if this FSM solution fits my concept of events. What about some contextual dialog parts? It seems that currently every time you start a dialog i looks exactly the same. What if one NPC gives you a task and you want to add one sentence to say but only if the task is done?